File: engine/model/src/main/scala/net/server/login/packet/AfterLoginPacket.scala
package net.server.login.packet

import net.server.MaplePacket

class AfterLoginPacket(private var _byte1: Byte, private var _byte2: Byte, private var _pin: String) extends MaplePacket {
  def byte1: Byte = _byte1

  def byte2: Byte = _byte2

  def pin: String = _pin

  def this(_byte1: Byte) {
    this(_byte1, 5, "")
  }

  def this(_byte1: Byte, _byte2: Byte) {
    this(_byte1, _byte2, "")
  }
}

File: engine/engine-model/src/main/java/net/server/login/packet/AfterLoginPacket.java
package net.server.login.packet;

import net.server.MaplePacket;

public record AfterLoginPacket(Byte byte1, Byte byte2, String pin) implements MaplePacket {
   public AfterLoginPacket(Byte byte1) {
      this(byte1, (byte) 5, "");
   }

   public AfterLoginPacket(Byte byte1, Byte byte2) {
      this(byte1, byte2, "");
   }
}
